### ISSUE TITLE
ES2022: arbitrary module namespace names

### ISSUE THREAD

**Issue by dnalborczyk** [rollup/rollup#4322]
check if `rollup` supports `arbitrary module names` after bumping `acorn`. _(not yet released)_

`acorn` PR: https://github.com/acornjs/acorn/pull/1090

Add ES2022 normative change estree/estree#269
Normative: Arbitrary module namespace identifier names tc39/ecma262#2154

**Comment by LongTengDao** [comment]
already released now (acorn v8.7.0)

**Comment by sapphi-red** [comment]
Rollup (v3.7.4) now uses acorn v8.8.0 but it throws the following error. ([repro](https://stackblitz.com/edit/rollup-template-3vkcu7?file=src%2Fmain.js))
```
TypeError: Cannot read properties of undefined (reading '0')
    at Module.getVariableForExportName (file:///home/projects/rollup-template-3vkcu7/node_modules/rollup/dist/shared/rollup.js:13595:17)
    at getVariableForExportNameRecursive (file:///home/projects/rollup-template-3vkcu7/node_modules/rollup/dist/shared/rollup.js:13281:19)
    at Module.traceVariable (file:///home/projects/rollup-template-3vkcu7/node_modules/rollup/dist/shared/rollup.js:13819:35)
    at ModuleScope.findVariable (file:///home/projects/rollup-template-3vkcu7/node_modules/rollup/dist/shared/rollup.js:12353:39)
    at Identifier.bind (file:///home/projects/rollup-template-3vkcu7/node_modules/rollup/dist/shared/rollup.js:8322:40)
    at CallExpression.bind (file:///home/projects/rollup-template-3vkcu7/node_modules/rollup/dist/shared/rollup.js:6113:28)
    at CallExpression.bind (file:///home/projects/rollup-template-3vkcu7/node_modules/rollup/dist/shared/rollup.js:9839:15)
    at ExpressionStatement.bind (file:///home/projects/rollup-template-3vkcu7/node_modules/rollup/dist/shared/rollup.js:6117:23)
    at Program.bind (file:///home/projects/rollup-template-3vkcu7/node_modules/rollup/dist/shared/rollup.js:6113:28)
    at Module.bindReferences (file:///home/projects/rollup-template-3vkcu7/node_modules/rollup/dist/shared/rollup.js:13451:18)
```
I guess rollup needs to change the code.

**Comment by lukastaegert** [comment]
Thanks for verifying this needs work. I had it somewhere deep in my todo list to check this again, but if someone wants to have a stab at fixing this, this would also be welcome.

**Comment by lukastaegert** [comment]
Fix at #4770
